src/main/java/org/aksw/simba/squirrel/analyzer/impl/MimeTypeDetector.java

```diff
@@ -0,0 +1,43 @@
+package org.aksw.simba.squirrel.analyzer.impl;
+
+import org.aksw.simba.squirrel.analyzer.TypeDetector;
+import org.apache.jena.riot.Lang;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+
+public class MimeTypeDetector implements TypeDetector {
+    private static final Logger LOGGER = LoggerFactory.getLogger(RDFAnalyzer.class);
+
+    public Lang detectMimeType(File data) {
+
+        Lang mimeType = null;
+        try {
+            FileInputStream inputStream = new FileInputStream(data.getAbsolutePath());
+            FiniteStateMachine machine1 = RdfAutomata.buildRDFStateMachine();
+            FiniteStateMachine machine2 = TurtleAutomata.buildTurtleStateMachine();
+            char current;
+            while (inputStream.available() > 0 && (!machine1.canStop() || !machine2.canStop())) {
+                current = (char) inputStream.read();
+                if(!machine1.isError()) //whichever leads to error state is processed anymore.
+                    machine1 = machine1.switchState(String.valueOf(current));
+                if(!machine2.isError())
+                    machine2 = machine2.switchState(String.valueOf(current));
+                System.out.println(current);
+            }
```

Comment:
The loop already goes into the correct direction but I still see a major problem and possible enhancements
1. You are still assuming that your finite state automatons will reach an end status (you are even assuming that both automatons will reach that although one of them won't be able to parse the file at all! :open_mouth:). But for that you will have to read the complete file! :disappointed: 
2. If a machine reaches an error state, you don't have to take it into account, anymore. It shouldn't be possible for the machine to leave this error state.
3. What we are interested in is the last machine that has no error. So I would suggest the following workflow: 
    1. create a list of machines based on the factories that this instance has (see comment above)
    2. read the single characters one by one and give it to the machines (as you are already doing it)
    3. if a machine can not process the character and ends up in an error state, remove it from the list
    4. repeat 2 and 3 until only one machine is left OR if there are no more characters available.